src/inmanta/deploy/scheduler.py

```diff
@@ -765,6 +765,7 @@ async def _new_version(
         new_versions: Sequence[ModelVersion],
         *,
         up_to_date_resources: Optional[Set[ResourceIdStr]] = None,
+        last_deploy_time: dict[ResourceIdStr, datetime.datetime] | None = None,
```

Comment:
I think we should add to the docstring, similar to that of `up_to_date_resources` that "it is only expected for the very first version when the scheduler is started with a fresh state".